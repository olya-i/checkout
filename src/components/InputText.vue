<script lang="ts">
import { defineComponent, type PropType } from "vue";

export default defineComponent({
  props: {
    label: String,
    name: String,
    modelValue: String as PropType<unknown>,
    isInvalid: Boolean,
    invalidMessage: String,
  },
  emits: ["update:modelValue", "validate"],
  computed: {
    value: {
      get() {
        return this.modelValue;
      },
      set(value: string) {
        this.$emit("update:modelValue", value);
        this.$emit("validate");
      },
    },
    placeholder(): string {
      return `Enter your ${this.label}`;
    },
  },
});
</script>

<template>
  <div class="input-wrapper">
    <label class="label">
      {{ label }}
      <input
        class="input"
        :class="{ error: isInvalid }"
        type="text"
        :name="name"
        v-model="value"
        :placeholder="placeholder"
      />
    </label>
    <div v-if="isInvalid" class="error-text">
      {{ invalidMessage }}
    </div>
  </div>
</template>

<style lang="scss" scoped>
.input-wrapper {
  margin-bottom: 20px;
}
.label {
  display: block;
  font-weight: bold;
  font-size: 20px;
}

.input {
  display: block;
  margin-top: 4px;
  padding: 12px 20px;
  border-radius: var(--border-radius--sm);
  transition: 0.2s ease-in;
  font-size: 20px;
  border: 1px solid var(--gray--dark);
  color: var(--dark);
  background: var(--white);
  width: 100%;

  &:hover {
    border-color: var(--blue);
  }

  &:focus {
    outline: 2px solid var(--blue--dark);
    border-color: var(--blue--dark);
  }

  &::placeholder {
    color: var(--gray--dark);
  }

  &.error {
    border-color: var(--red);
    outline-color: var(--red);
  }
}

.error-text {
  color: var(--red);
  font-size: 20px;
}
</style>
