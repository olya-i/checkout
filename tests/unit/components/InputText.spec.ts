import InputText from "../../../src/components/InputText.vue";
import { mount } from "@vue/test-utils";

describe("Mounted InputText", () => {
  it("renders correctly", () => {
    const wrapper = mount(InputText, {
      props: {
        label: "Input label",
        name: "input",
        modelValue: "some value",
        isInvalid: false,
        invalidMessage: "Some error",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
