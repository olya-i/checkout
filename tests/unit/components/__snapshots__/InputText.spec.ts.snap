// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted InputText renders correctly 1`] = `
<div
  class="input-wrapper"
>
  <label
    class="label"
  >
    Input label 
    <input
      class="input"
      name="input"
      placeholder="Enter your Input label"
      type="text"
    />
  </label>
  <!--v-if-->
</div>
`;
